pip/utils/hashes.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+
+import hashlib
+
+from pip.exceptions import (HashMismatch, HashMissing, InstallationError,
+                            FAVORITE_HASH)
+from pip._vendor.six import iteritems, iterkeys, itervalues
+
+
+class Hashes(object):
+    """A wrapper that builds multiple hashes at once and checks them against
+    known-good values
+
+    """
+    def __init__(self, hashes=None):
+        """
+        :param hashes: A dict of algorithm names pointing to lists of allowed
+            hex digests
+        """
+        self._goods = {} if hashes is None else hashes
+
+    def check_against_chunks(self, chunks):
+        """Check good hashes against ones built from iterable of chunks of
+        data.
+
+        Raise HashMismatch if none match.
+
+        """
+        gots = {}
```

Comment:
"goods" isn't proper English either. :-) I suppose I did this for symmetry. I could change them to "allowed" and "got" or something if it reads better to you.